core/src/main/java/org/mule/registry/AbstractRegistryBroker.java

```diff
@@ -169,68 +227,111 @@ public boolean isRemote()
 
         return results;
     }
-    
+
+    @Override
     public <T> Collection<T> lookupObjectsForLifecycle(Class<T> type)
     {
-        Collection<T> objects = new ArrayList<T>();
+        Collection<Registry> registries = getRegistries();
+        if (registries.isEmpty())
+        {
+            return limbo.lookupObjectsForLifecycle(type);
+        }
 
-        Iterator it = getRegistries().iterator();
-        while (it.hasNext())
+        Collection<T> objects = new ArrayList<>();
+        for (Registry registry : registries)
         {
-            objects.addAll(((Registry) it.next()).lookupObjectsForLifecycle(type));
+            objects.addAll(registry.lookupObjectsForLifecycle(type));
         }
+
         return objects;
     }
 
-       
+    /**
+     * {@inheritDoc}
+     */
+    @Override
     public void registerObject(String key, Object value) throws RegistrationException
     {
-        registerObject(key, value, null);
-    }
-
-    public void registerObject(String key, Object value, Object metadata) throws RegistrationException
```

Comment:
That we used it only in some test it does not mean that some user can be relying on it. This  change must be properly documented